src/platform/plugins/shared/discover/public/application/main/state_management/redux/internal_state.ts

```diff
@@ -348,6 +350,31 @@ export const internalStateSlice = createSlice({
       state.persistedDiscoverSession = action.payload.persistedDiscoverSession;
     });
 
+    builder.addCase(initializeSingleTab.pending, (state, action) =>
+      withTab(state, action.meta.arg, (tab) => {
+        tab.initializationState = { initializationStatus: TabInitializationStatus.InProgress };
+      })
+    );
+
+    builder.addCase(initializeSingleTab.fulfilled, (state, action) =>
+      withTab(state, action.meta.arg, (tab) => {
+        tab.initializationState = {
+          initializationStatus: action.payload.showNoDataPage
+            ? TabInitializationStatus.NoData
+            : TabInitializationStatus.Complete,
+        };
+      })
+    );
+
+    builder.addCase(initializeSingleTab.rejected, (state, action) =>
+      withTab(state, action.meta.arg, (tab) => {
+        tab.initializationState = {
+          initializationStatus: TabInitializationStatus.Error,
+          error: action.error,
+        };
+      })
+    );
```

Comment:
This will unfortunately put `SavedObjectNotFound` into the Redux store which isn't serializable (and partially why I originally handled it in the component). But the multiple initializations bug is worse, and there's not a great alternative with how `RedirectWhenSavedObjectNotFound` currently works.

We should change the redirect logic at some point to get rid of the `SavedObjectNotFound` weirdness, which would fix this too.